app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php

```diff
@@ -77,4 +81,168 @@ public function indexAction()
     {
         $this->_initAction()->renderLayout();
     }
+
+    /**
+     * Download requested log file
+     */
+    public function downloadAction()
+    {
+        $params   = Mage::app()->getRequest()->getParams();
+        $fileName = Mage::getBaseDir('var') . DS . 'log' . DS . $params['name'];
+
+        $this->varien_io = new Varien_Io_File();
+        $this->missingFile($fileName, $params['name']);
+
+        if (isset($params['rows']) && $this->varien_io->fileExists($fileName) && @is_readable($fileName)) {
+            $file  = $this->getTemporaryFile($fileName, $params['rows']);
+            $lines = count($file);
+
+            if ($lines < $params['rows']) {
+                $this->downloadFullFile($fileName);
+            } else {
+                $pointer = $lines - $params['rows'];
+                $text = "";
+                for (; $lines > $pointer; $pointer++) {
+                    $text .= $file[$pointer];
+                }
+                $this->setHeaders($fileName);
+                echo $text;
+            }
+        } elseif ($this->varien_io->fileExists($fileName) && @is_readable($fileName)) {
+            $this->downloadFullFile($fileName);
+        }
+    }
+
+    /**
+     * Delete requested log file
+     */
+    public function deleteAction()
+    {
+        $params = Mage::app()->getRequest()->getParams();
+        $fileName = Mage::getBaseDir('var') . DS . 'log' . DS . $params['name'];
+
+        $this->varien_io = new Varien_Io_File();
+        $this->missingFile($fileName, $params['name']);
+
+        if ($this->varien_io->fileExists($fileName) && @is_readable($fileName)) {
+            $oystLog = Oyst_OneClick_Helper_Data::MODULE_NAME . '.log';
+            if ($params['name'] !== $oystLog) {
+                Mage::getSingleton('core/session')->addError(
+                    Mage::helper('oyst_oneclick')->__('Cannot delete "%s" file.', $params['name'])
+                );
+            } else {
+                $this->varien_io->rm($fileName);
+                Mage::getSingleton('core/session')->addSuccess(
+                    Mage::helper('oyst_oneclick')->__('Successfully deleted "%s".', $oystLog)
+                );
+            }
+        }
+        $this->_redirectReferer();
+    }
+
+    /**
+     * Display error message if cannot find file.
+     *
+     * @param $filePath
+     * @param $name
+     */
+    private function missingFile($filePath, $name)
+    {
+        if (!$this->varien_io->fileExists($filePath)) {
+            Mage::getSingleton('core/session')->addError(
+                Mage::helper('oyst_oneclick')->__('File "%s" does not exist.', $name)
+            );
+        }
+    }
+
+    /**
+     * Set download headers.
+     *
+     * @param $file
+     */
+    private function setHeaders($file, $name = null)
```

Comment:
@rus-daniel I let you do a refactoring of the download method.

You can inspire of:
```
$fileName = Mage::getBaseDir('var') . DS . 'log' . DS . 'Oyst_OneClick.log';

$this->_prepareDownloadResponse(
    'Oyst_OneClick.log',
    array(
        'type' => 'filename',
        'value' => $fileName,
    )
);

$this->getResponse()->sendHeaders();
```